src/components/pagination/index.tsx

```diff
@@ -0,0 +1,112 @@
+};
+
+export interface Props {
+  count: number;
+  page: number;
+  siblingCount?: number;
+  onChange: (page: number) => void;
+}
+
+const Pagination: FC<PropsWithChildren<Props>> = (props) => {
+  const { onChange, count, page, siblingCount = 1 } = props;
+  const { paginationContainer, dots } = useMultiStyleConfig('Pagination');
+
+  const paginationRange = useMemo(() => {
+    // Default number of page buttons: firstPage + lastPage + currentPage + left side dots + right side dots
+    const pageButtonsCount = 5;
+    const totalPageButtonsCount = siblingCount + pageButtonsCount;
+
+    if (totalPageButtonsCount >= count) {
+      return range(1, count);
+    }
+
+    const leftSiblingIndex = Math.max(page - siblingCount, 1);
+    const rightSiblingIndex = Math.min(page + siblingCount, count);
+
+    const shouldShowLeftDots = leftSiblingIndex > 2;
+    const shouldShowRightDots = rightSiblingIndex < count - 2;
+
+    const firstPageIndex = 1;
+    const lastPageIndex = count;
+
+    if (!shouldShowLeftDots && shouldShowRightDots) {
+      const leftItemCount = 3 + 2 * siblingCount;
+      const leftRange = range(1, leftItemCount);
+
+      return [...leftRange, Dots, count];
+    }
+
+    if (shouldShowLeftDots && !shouldShowRightDots) {
+      const rightItemCount = 3 + 2 * siblingCount;
+      const rightRange = range(count - rightItemCount + 1, count);
+      return [firstPageIndex, Dots, ...rightRange];
+    }
+
+    if (shouldShowLeftDots && shouldShowRightDots) {
+      const middleRange = range(leftSiblingIndex, rightSiblingIndex);
+      return [firstPageIndex, Dots, ...middleRange, Dots, lastPageIndex];
+    }
+
+    return [];
+  }, [count, siblingCount, page]);
+
+  if (page === 0 || paginationRange.length < 2) {
+    return null;
+  }
+
+  const onNext = () => onChange(page + 1);
+  const onPrevious = () => onChange(page - 1);
+
+  const isLastPage = page === paginationRange[paginationRange.length - 1];
+
+  return (
+    <Box __css={paginationContainer}>
+      <PaginationButton
+        isDisabled={page === 1}
+        onClick={onPrevious}
+        data-testid="prev-button"
+      >
+        <ChevronLeftSmallIcon color={page === 1 ? 'gray.300' : 'gray.900'} />
+      </PaginationButton>
+      {paginationRange.map((pageNumber, index) => {
+        if (pageNumber === Dots) {
+          return (
+            <Box __css={dots} key={`paginationButton-${index}`}>
+              &#8230;
+            </Box>
+          );
+        }
+
+        return (
+          <PaginationButton
+            key={`paginationButton-${index}`}
+            isActive={pageNumber === page}
+            onClick={() => onChange(pageNumber as number)}
+          >
+            {pageNumber}
+          </PaginationButton>
+        );
+      })}
+      <PaginationButton
+        isDisabled={isLastPage}
+        onClick={onNext}
+        data-testid="next-button"
+      >
+        <ChevronRightSmallIcon color={isLastPage ? 'gray.300' : 'gray.900'} />
```

Comment:
![Bildschirmfoto 2022-11-17 um 08 22 05](https://user-images.githubusercontent.com/71456764/202381907-886c9b82-2f7c-4c96-b7a8-d5261956f693.png)
could you check the alignment? it looks not centered